fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -363,6 +421,27 @@ def add_text(
     for i in range(num_sides):
         if "deluxe" in states[i].model_name:
             hint_msg = SLOW_MODEL_MSG
+    
+    if file_extension == ".pdf":
+        document_text = llama_parse(files[0])
```

Comment:
Ideally, we abstract pdf parser here. We should call it something like `pdf_parse` and by default it uses llamaparse, but we can switch to others when needed.